lib/rex/post/meterpreter/extensions/stdapi/sys/process_subsystem/memory.rb

```diff
@@ -139,7 +139,10 @@ def search(needles, min_search_len = 5, match_len = 500)
     request = Packet.create_request(COMMAND_ID_STDAPI_SYS_PROCESS_MEMORY_SEARCH)
 
     request.add_tlv(TLV_TYPE_PID, process.pid)
-    needles.each { | needle | request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE, needle) }
+    needles.each do | needle |
+      request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE, needle)
+      request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE_LEN, needle.bytesize)
```

Comment:
It seems odd that we need this additional information, when the length value is already baked into the TLV (Type **Length** Value) itself

I wonder i this is being added as a workaround for your current Meterpreter implementation? 🤔